### PR TITLE
💫 STAR Solo multi-map + New new new QC

### PR DESCRIPTION
**Pull request by migbro**
## Description

 - Added ability incorporate multi-mapping gene counts (https://github.com/alexdobin/STAR/blob/master/docs/STARsolo.md#multi-gene-reads)
 - Created new QC script based on [this tutorial](https://github.com/hbctraining/scRNA-seq_online/blob/scRNAseq/lessons/04_SC_quality_control.md). 10X_SEURAT_HBC_SCRNA_QC.md doc better summarizes QC outputs

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [Classic PBMC task](https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/6f949ca6-da33-4741-9f4b-ba1c3da86e54/)
- [ ] [Project Hope Sample](https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/a93ca735-e5c0-4798-bd1b-c3dfba0cb227/)

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

File: tools/convert_to_h5.cwl [hunk 1]
```diff
@@ -1,44 +1,62 @@
...
+        var raw_dir = inputs.solo_counts_dir.path + "/" + inputs.soloFeatures + "/raw/";
+        var setup_raw_cmd = "mkdir RAW && cp " + raw_dir + "*.tsv RAW/ &&";
+        if (inputs.raw_count_choice == "Unique" || inputs.raw_count_choice == null){
+          setup_raw_cmd +=  " cp " + raw_dir + "matrix.mtx RAW/ &&"
+        }
+        else{
+          setup_raw_cmd += " cp " + raw_dir + "UniqueAndMult-" + inputs.raw_count_choice + ".mtx RAW/matrix.mtx &&";
+        }
+        return setup_raw_cmd;
+      }
```
Comment (dmiller15) [inline tools/convert_to_h5.cwl:37]: Just having fun
```suggestion
    valueFrom: >-
      mkdir RAW
      && cp $(inputs.solo_counts_dir.path)/$(inputs.soloFeatures)/raw/*.tsv RAW/ 
      && cp $(inputs.solo_counts_dir.path)/$(inputs.soloFeatures)/raw/$(["Unique",null].indexOf(inputs.raw_count_choice) != -1 ? "matrix" : "UniqueAndMult-" + inputs.raw_count_choice).mtx RAW/matrix.mtx &&
```

File: workflows/kf_STAR_Solo_10x_alignment_wf.cwl [hunk 2]
```diff
@@ -147,10 +152,10 @@ inputs:
       gene, matching CellRanger > 3.0.0. Only works with -soloUMIdedup 1MM_CR", default: "MultiGeneUMI_CR"}
   soloUMIdedup: {type: ['null', {type: enum, name: soloUMIdedup, symbols: ["1MM_All", "1MM_Directional_UMItools", "1MM_Directional",
           "Exact", "NoDedup", "1MM_CR"]}], doc: "type of UMI deduplication (collapsing) algorithm). 1MM_All: all UMIs with 1 mismatch
-      distance to each other are collapsed (i.e. counted once). 1MM_Directional_UMItools: follows the \u201Ddirectional\u201D method
-      from the UMI-tools by Smith, Heger and Sudbery (Genome Research 2017). 1MM_Directional: same as 1MM Directional UMItools, but
-      with more stringent criteria for duplicate UMIs. Exact: only exactly matching UMIs are collapsed NoDedup: no deduplication of
-      UMIs, count all reads. 1MM_CR: CellRanger2-4 algorithm for 1MM UMI collapsing.", default: "1MM_CR"}
+      distance to each other are collapsed (i.e. counted once). 1MM_Directional_UMItools: follows the ”directional” method from the
```
Comment (dmiller15) [inline workflows/kf_STAR_Solo_10x_alignment_wf.cwl:155]: curly quotes bad
```suggestion
      distance to each other are collapsed (i.e. counted once). 1MM_Directional_UMItools: follows the 'directional' method from the
```

**Review by dmiller15 — APPROVED** [review]
Mostly cosmetic stuff

File: tools/convert_to_h5.cwl (hunk 1 above)
Comment (migbro) [reply tools/convert_to_h5.cwl:37]: Hmmm, I'll try it out tomorrow morning. I think I am tapped out for the day

**Review by sickler-alex — APPROVED** [review]

**Comment by migbro** [comment]
Gave @dmiller15 's code cleanup commit a test, and worked well: https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/2267b2bb-c1d8-4bcc-96fb-e34fb2295e44

**Pull request merged**